### PR TITLE
Remove `Query\Builder::__constructor` overload

### PR DESCRIPTION
**Pull request by GromNaN**
The constructor don't need to be overloaded to set the default grammar. It is read from the connection by default.

https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Query/Builder.php#L246

### CODE REVIEW

**Review by alcaeus — APPROVED** [review]

**Review by jmikola — APPROVED** [review]
I didn't see answers to my original questions in https://github.com/GromNaN/laravel-mongodb/pull/26, so please just take a look at those before merging.

LGTM if my assumptions in the previous questions were correct.

Everything here LGTM.

**Comment by alcaeus** [comment]
@GromNaN LGTM. Can you rebase to resolve the conflicts in the changelog?

**Comment by GromNaN** [comment]
Rebased.

**Review by alcaeus — APPROVED** [review]

**Pull request merged**
